### PR TITLE
Fix a group of mixer attributes generated from k8s when pod name contains dot

### PR DESCRIPTION
**Pull request by zhranklin**
Fix a group of mixer attributes generated from k8s env such as source/destination's name, owner, namespace when dot in pod name

### CODE REVIEW

**Comment by zhranklin** [comment]
fix: #20028

**Comment by zhranklin** [comment]
@googlebot I fixed it.

File: mixer/adapter/kubernetesenv/kubernetesenv.go [hunk 1]
```diff
@@ -266,16 +266,15 @@ func (h *handler) findPod(uid string) (cacheController, *v1.Pod, bool) {
...
 		return uid
 	}
 	fullname := strings.TrimPrefix(uid, kubePrefix)
-	if strings.Contains(fullname, ".") {
-		parts := strings.Split(fullname, ".")
-		if len(parts) == 2 {
-			return key(parts[1], parts[0])
-		}
+	dotPos := strings.LastIndex(fullname, ".")
+	if dotPos != -1 {
```
Comment (mandarjog) [inline mixer/adapter/kubernetesenv/kubernetesenv.go]: What happens when dotPos == 0 ?

Comment (nrjpoddar) [reply mixer/adapter/kubernetesenv/kubernetesenv.go]: that's not a valid k8s name afaik.

**Comment by mandarjog** [comment]
We are not adding any new features to Mixer now, but given that this is a Bug fix we can add it to 1.4.

**Comment by nrjpoddar** [comment]
@mandarjog even though this PR fixes this metric related issue, overall Istio code base has lot of assumptions around K8s resources not having a ".".  K8s used to not allow "." in names in earlier releases and I think in other places we still assume "." is a separator for namespace.

**Comment by zhranklin** [comment]
@howardjohn maybe you mean #19022
This is to fix the mixer plugin config that pilot give to envoy and correct the `source.namespace`. But some other attributes generated by apa `kubernetesenv` are still wrong.

**Comment by zhranklin** [comment]
@douglas-reid Ok, I will add some test code.

**Comment by zhranklin** [comment]
@googlebot I fixed it.

**Pull request merged**